plugin/storage/es/factory.go

```diff
@@ -162,6 +163,7 @@ func createSpanReader(
 		UseReadWriteAliases:           cfg.UseReadWriteAliases,
 		Archive:                       archive,
 		RemoteReadClusters:            cfg.RemoteReadClusters,
+		Tracer:                        otel.GetTracerProvider().Tracer("span-reader"),
```

Comment:
```suggestion
		Tracer:                        otel.GetTracerProvider().Tracer("esSpanStore.SpanReader"),
```

Reply:
also, this should be split: here you only pass `Tracer:  otel.GetTracerProvider()`, and `esSpanStore.NewSpanReader()` calls `.Tracer()` on it. Because in the future it may need Provider for other purposes, you don't know. Provider is the shared object, Tracers are created by individual instrumentations as needed.